site/app/controllers/MiscController.php

```diff
@@ -24,59 +24,110 @@ public function run() {
         }
     }
 
-    private function displayFile() {
-        //Additional security
-        if (!($this->core->getUser()->accessGrading())) {
-            throw new \InvalidArgumentException("It does not look like you're allowed to access this page.");
-        }
-        foreach (explode(DIRECTORY_SEPARATOR, $_REQUEST['path']) as $part) {
-            if ($part == ".." || $part == ".") {
-                throw new \InvalidArgumentException("Cannot have a part of the path just be dots");
+    // function to check that this is a valid access request
+    private function checkValidAccess($is_zip, $dir="",$path="",$user_id="") {
```

Comment:
instead of having this function throw exceptions (which I admit was a clumsy mistake on my part initially), I'd rather see this function return true or false.